From: Michal Privoznik <mprivozn@redhat.com>
Date: Fri, 10 Oct 2025 15:20:05 +0200
Subject: wireshark: Switch header files to #pragma once

The genxdrstub.pl script generates some header files. But they
use the old pattern to guard against multiple inclusion:

  #ifndef SOMETHING_H
  #define SOMETHING_H
  ...
  #endif

Change the script to generate just '#pragma once' used everywhere
else in our code.

Signed-off-by: Michal Privoznik <mprivozn@redhat.com>
Reviewed-by: Peter Krempa <pkrempa@redhat.com>
(cherry picked from commit 41d3b457972bde85991fa7ed6f282370aca4b2af)

Bug-Debian: https://bugs.debian.org/1118069

Forwarded: not-needed
Origin: https://gitlab.com/libvirt/libvirt/-/commit/41d3b457972bde85991fa7ed6f282370aca4b2af
---
 tools/wireshark/util/genxdrstub.pl | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/tools/wireshark/util/genxdrstub.pl b/tools/wireshark/util/genxdrstub.pl
index 8cfda25..01b663a 100755
--- a/tools/wireshark/util/genxdrstub.pl
+++ b/tools/wireshark/util/genxdrstub.pl
@@ -563,11 +563,8 @@ sub add_header_file {
     local $self->{header_contents} = [];
     $self->print("/* *DO NOT MODIFY* this file directly.\n");
     $self->print(" * This file was generated by $0 from libvirt version $libvirt_version */\n");
-    my $ucname = uc $name;
-    $self->print("#ifndef _$ucname\_H_\n");
-    $self->print("#define _$ucname\_H_\n");
+    $self->print("#pragma once\n");
     $block->();
-    $self->print("#endif /* _$ucname\_H_ */");
     push @{ $self->{headers} }, [ $name, delete $self->{header_contents} ];
 }
 
